OutOfSchool/OutOfSchool.AuthCommon/Services/Password/PasswordGenerator.cs

```diff
@@ -1,65 +1,50 @@
-﻿namespace OutOfSchool.AuthCommon.Services.Password;
+﻿using System.Security.Cryptography;
+
+namespace OutOfSchool.AuthCommon.Services.Password;
 
 public static class PasswordGenerator
 {
     /// <summary>
-    /// Generates a Random Password
-    /// respecting the given strength requirements.
+    /// Generates a Random Password.
     /// </summary>
-    /// <param name="passwordOptions">A valid PasswordOptions object
-    /// containing the password strength requirements.</param>
     /// <returns>A random password.</returns>
-    public static string GenerateRandomPassword(PasswordOptions passwordOptions)
+    public static string GenerateRandomPassword()
     {
-        string[] randomChars = new[]
-        {
-            "ABCDEFGHJKLMNOPQRSTUVWXYZ",    // uppercase
-            "abcdefghijkmnopqrstuvwxyz",    // lowercase
-            "0123456789",                   // digits
-            "!@$?_-",                       // non-alphanumeric
-        };
+        string[] allowedCharSets =
+        [
+            "ABCDEFGHJKLMNOPQRSTUVWXYZ",
+            "abcdefghijkmnopqrstuvwxyz",
+            "0123456789",
+            Constants.ValidationSymbols,
+        ];
 
-        CryptoRandom rand = new CryptoRandom();
+        List<char> password = [];
 
-        List<char> chars = new List<char>();
-
-        if (passwordOptions.RequireUppercase)
+        foreach (var charSet in allowedCharSets)
         {
-            chars.Insert(
-                rand.Next(0, chars.Count),
-                randomChars[0][rand.Next(0, randomChars[0].Length)]);
+            password.Insert(
+                GetRandomInt32(password.Count),
+                charSet[GetRandomInt32(charSet.Length)]);
         }
 
-        if (passwordOptions.RequireLowercase)
+        while (password.Count < Constants.PasswordMinLength)
         {
-            chars.Insert(
-                rand.Next(0, chars.Count),
-                randomChars[1][rand.Next(0, randomChars[1].Length)]);
+            var randomCharSetIndex = GetRandomInt32(allowedCharSets.Length);
+            var randomChar =
+                allowedCharSets[randomCharSetIndex][
+                GetRandomInt32(allowedCharSets[randomCharSetIndex].Length)];
```

Comment:
(Computational hardness) Here you loose the true "randomness". Now you are taking a random character set and then taking random character in the set. 

Imagine, that you have 2 sets: the very small one (1 character in extreme) and the very big one (100 characters). In your scenario - that 1 character from the first set would be taken with 50% probablity, while characters from bigger set would be taken with 0.5% probability. This is pretty different probability.

Please consider concatenating all sets into 1 string and taking random char from it.

Reply:
I agree with your argument, but in this case the password must include at least one character from each set.

Reply:
> the password must include at least one character from each set.

... and you get this in `foreach` loop. That's why there is no necessity to do this here in `while` loop. Moreover, due to randomness theer is (small) probability to get same charset for one password generation in this `while` loop, so your argument is bit irrelevant. 

Please reconsider your comment.

Reply:
You are right. I fixed this in my latest commit. Thank you!